LibWeb: Layout standalone SVG file obeys width/height attributes

manuel-za opened the pull request:
Fixes #2668 

- parity with Chromium (131.0.6778.85)
- very close to Firefox (129.0.2) -- difference not material

circl-lastname commented:
Could you write "Fixes #2668"? That way Github automatically marks that issue resolved once merged

AtkinsSJ commented on Libraries/LibWeb/Layout/SVGFormattingContext.cpp:
```diff
@@ -184,6 +184,21 @@ void SVGFormattingContext::run(AvailableSpace const& available_space)
+        if (svg_box_state.node().computed_values().height().is_length())
+            svg_box_state.set_content_height(svg_box_state.node().computed_values().height().length().absolute_length_to_px());
+
+        // FIXME: find a more precise way to determine whether we are rendering a standalone SVG document
+        // Will break for image-SVG elements without a specified viewBox included in an XML document
+        // DOM::Document::is_xml_document basically means "it is not an html document" (https://dom.spec.whatwg.org/#xml-document)
```
Looks like for SVG documents we also set the content type, so you should be able to check `content_box().document().content_type() == "image/svg+xml"_string`.

Also, `<svg>` can appear inside another `<svg>`. If you want to only set the width and height for top-level SVGs, you'll need to add a check for that somehow.

AtkinsSJ reviewed (changes requested):
Hi, welcome to Ladybird! :^)

A couple of more general comments:
- That first commit adds a test before the commit that makes the test pass. Please move it either into the second commit, or after it - all tests should pass when they're added.
- Commit titles should be imperatives (like a command). So, "Add unit tests..." instead of just "Unit tests...", and "Make standalone SVG files obey width..."

manuel-za commented:
@AtkinsSJ : Thanks for the comments. Makes sense.
Rather than rewrite history + pollute the log, I created another PR addressing the comments: https://github.com/LadybirdBrowser/ladybird/pull/2806

I believe this new PR addresses your requested changes:
- unit tests come after the code change
- found a reliable way to detect running in the top formatting context

I would appreciate a review. Thanks! :]m

manuel-za commented:
...closing and moving to #2806

manuel-za commented:
@AtkinsSJ : Thanks for the comments. Makes sense.
Rather than rewrite history + pollute the log, I created another PR addressing the comments: https://github.com/LadybirdBrowser/ladybird/pull/2806

I believe this new PR addresses your requested changes:
- unit tests come after the code change
- found a reliable way to detect running in the top formatting context

I would appreciate a review. Thanks! :]m
